spring-batch-docs/asciidoc/retry.adoc

```diff
@@ -210,70 +210,73 @@ Be very careful with the implementation of
+// `RetryContextCache` into `RetryTemplate` and passing the `RecoveryCallback` to
+// `RetryOperations`
 
 [[retryPolicies]]
 
 
 === Retry Policies
 
-Inside a `RetryTemplate` the decision to retry
+Inside a `RetryTemplate`, the decision to retry
     or fail in the `execute` method is determined by a
-    `RetryPolicy` which is also a factory for the
+    `RetryPolicy`, which is also a factory for the
     `RetryContext`. The
     `RetryTemplate` has the responsibility to use the
     current policy to create a `RetryContext` and pass
     that in to the `RetryCallback` at every attempt.
-    After a callback fails the `RetryTemplate` has to
+    After a callback fails, the `RetryTemplate` has to
     make a call to the `RetryPolicy` to ask it to update
-    its state (which will be stored in the
-    `RetryContext`), and then it asks the policy if
-    another attempt can be made. If another attempt cannot be made (e.g. a
+    its state (which is stored in the
+    `RetryContext`) and then asks the policy if
+    another attempt can be made. If another attempt cannot be made (such as when a
     limit is reached or a timeout is detected) then the policy is also
-    responsible for handling the exhausted state. Simple implementations will
-    just throw `RetryExhaustedException` which will cause
+    responsible for handling the exhausted state. Simple implementations
+    throw `RetryExhaustedException`, which causes
     any enclosing transaction to be rolled back. More sophisticated
     implementations might attempt to take some recovery action, in which case
     the transaction can remain intact.
 
 
 [TIP]
 ====
-Failures are inherently either retryable or not - if the same
+Failures are inherently either retryable or not. If the same
       exception is always going to be thrown from the business logic, it
-      doesn't help to retry it. So don't retry on all exception types - try to
-      focus on only those exceptions that you expect to be retryable. It's not
+      does no good to retry it. So do not retry on all exception types. Rather, try to
+      focus on only those exceptions that you expect to be retryable. It is not
       usually harmful to the business logic to retry more aggressively, but
-      it's wasteful because if a failure is deterministic there will be time
-      spent retrying something that you know in advance is fatal.
+      it is wasteful, because, if a failure is deterministic, you spend time
+      retrying something that you know in advance is fatal.
 ====
 
 
 Spring Batch provides some simple general purpose implementations of
-    stateless `RetryPolicy`, for example a
-    `SimpleRetryPolicy`, and the
-    `TimeoutRetryPolicy` used in the example
-    above.
+    stateless `RetryPolicy`, such as
+    `SimpleRetryPolicy` and
+    `TimeoutRetryPolicy` (used in the preceding example).
 
-The `SimpleRetryPolicy` just allows a retry on
+The `SimpleRetryPolicy`  allows a retry on
```

Comment:
nitpicky but 2 spaces.

Reply:
Nitpicky is what I do. One space, please. Good catch.